lib/nanoc/base/entities/configuration.rb

```diff
@@ -51,6 +59,21 @@ def with_defaults
       self.class.new(new_wrapped)
     end
 
+    def with_environment
+      return self unless @wrapped.key?(ENVIRONMENTS)
+
+      # Set active environment
+      env = @env || ENV.fetch('NANOC_ENV', 'default')
```

Comment:
The name `env` (and `@env`) is confusing; the name makes it sound like it’s similar to `ENV`. Maybe `env_name`?